HiP-DataStore/Core/WriteModel/EntityIndex.cs

```diff
@@ -78,7 +78,9 @@ public IReadOnlyCollection<int> AllIds(ResourceType entityType, ContentStatus st
                     .AsQueryable()
                     .FilterIf(!isAllowedGetAll, x =>
                         ((status == ContentStatus.All) && (x.Value.Status == ContentStatus.Published)) || (x.Value.UserId == userId))
-                    .Where(x => status == ContentStatus.All || x.Value.Status == status)
+                    .FilterIf(status == ContentStatus.All && !UserPermissions.IsAllowedToGetDeleted(User),
```

Comment:
I think these `FilterIf(...)`-calls could be integrated into the `FilterByStatus(...)` extension method